src/hotspot/share/gc/shenandoah/shenandoahMark.hpp

```diff
@@ -57,7 +57,7 @@ class ShenandoahMark: public StackObj {
 
 public:
   template<class T, ShenandoahGenerationType GENERATION>
-  static inline void mark_through_ref(T* p, ShenandoahObjToScanQueue* q, ShenandoahObjToScanQueue* old_q, ShenandoahMarkingContext* const mark_context, bool weak);
+  static inline bool mark_through_ref(T* p, ShenandoahObjToScanQueue* q, ShenandoahObjToScanQueue* old_q, ShenandoahMarkingContext* const mark_context, bool weak);
```

Comment:
document the semantics of the method's return value here and everywhere where a bool return replaces a previously void return.